test/test_inference.jl

```diff
@@ -33,17 +33,17 @@ end
         @test_inferred foldl(+, Map(exp) |> Filter(x -> x > 0), xs)
         @test_inferred foldl(+, Map(exp) |> Scan(+), xs)
         @test_inferred foldl(+, TakeLast(1), xs)
-        @test_inferred foldl(+, PartitionBy(identity) |> Map(first), xs)
+        @test_broken_inferred foldl(+, PartitionBy(identity) |> Map(first), xs)
         @test_inferred foldl(+, Unique(), xs)
-        @test_inferred foldl(right, TeeZip(Filter(isodd) |> Map(inc)), xs)
+        @test_broken_inferred foldl(right, TeeZip(Filter(isodd) |> Map(inc)), xs)
```

Comment:
Calling `complete` in `__foldl__` rescues type inference of the above two cases.